llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -209,6 +209,15 @@ static cl::opt<bool> EnableLoopFlatten("enable-loop-flatten", cl::init(false),
                                        cl::Hidden,
                                        cl::desc("Enable the LoopFlatten Pass"));
 
+// Experimentally allow loop header duplication. This should allow for better
```

Comment:
FWIW I don't think allowing loop-rotation for all loops to simplify some to memsets is the best way forward here, as I don't see any path towards having this enabled by default.

> If this ends up being profitable, we should drop this flag and
>  // making a code gen option that can be controlled independent of the opt level
> // and exposed through clang.

It is not clear to me what profitable means here?

Reply:
I think the new wording should be less ambiguous. Happy to change it if you have a preference, though.